src/main/java/org/rumbledb/api/Item.java

```diff
@@ -546,4 +546,13 @@ public Item modulo(Item other) {
     public Item idivide(Item other) {
         throw new UnsupportedOperationException("Operation not defined");
     }
+
+    /**
+     * Returns the dynamic type of the item (for error message purposes).
+     * 
+     * @return the dynamic type as an item type.
+     */
+    public ItemType getDynamicType() {
+        throw new UnsupportedOperationException("Operation not defined");
+    }
```

Comment:
I really like the precision of your name choice here with the 'dynamicType'. It was a fun exercise for me to evaluate the dynamic vs static nature of types in this context.